File: code-coaches/medium/hex-color-code-generator/hex_color_code_generator.cpp
#include <iostream>
#include <iomanip>
#include <sstream>

std::string rgb_to_hex(int r, int g, int b)
{
    // Convert each RGB value to a two-digit hex string
    std::stringstream ss;
    ss << '#' << std::hex << std::setfill('0')
       << std::setw(2) << r
       << std::setw(2) << g
       << std::setw(2) << b;

    // Get the hex code string from the stringstream
    std::string hex_code = ss.str();
    return hex_code;
}

int main()
{
    int r_rgb, g_rgb, b_rgb;
    std::cin >> r_rgb >> g_rgb >> b_rgb;

    std::cout << rgb_to_hex(r_rgb, g_rgb, b_rgb);

    return 0;
}

File: code-coaches/medium/hex-color-code-generator/HexColorCodeGenerator.java
import java.util.Scanner;

public class HexColorCodeGenerator {

    public static String rgbToHex(int r, int g, int b) {
        // Convert each RGB value to a two-digit hex string
        String hexCode = String.format("#%02x%02x%02x", r, g, b);
        return hexCode;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int rRgb = scanner.nextInt();
        int gRgb = scanner.nextInt();
        int bRgb = scanner.nextInt();

        System.out.println(rgbToHex(rRgb, gRgb, bRgb));

        scanner.close();
    }

}
